crates/sparrow-instructions/src/evaluators/list/collect_struct.rs

```diff
@@ -395,16 +419,142 @@ impl CollectStructEvaluator {
         );
 
         // Now update the new state using the last entity take indices
-        let new_state = update_token_state(entity_take_indices, old_state_flat, input, fields)?;
+        let new_state = update_token_state(&entity_take_indices, old_state_flat, input, fields)?;
         token.set_state(Arc::new(new_state));
 
         Ok(Arc::new(result))
     }
+
+    /// Evaluates the collect function for structs with a `trailing` window.
+    ///
+    /// State is handled in order of "update -> emit -> reset".
+    ///
+    /// Follows the same implementation as above, but includes values in [current time - duration].
+    fn evaluate_trailing_windowed(
```

Comment:
I wonder if there is a way to just use the existing logic. This is probably fine for now, but we're getting pretty duplicated. Really want to see if we can have better aggregation code in the partitioned path.